inc/ThirdParty/Hostings/HostResolver.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace WP_Rocket\ThirdParty\Hostings;
+
+/**
+ * Host Resolver.
+ *
+ * @since 3.6.3
+ */
+class HostResolver {
+
+	/**
+	 * Name of the current host service.
+	 *
+	 * @var string
+	 */
+	private static $hostname = '';
+
+	/**
+	 * Get the name of an identifiable hosting service.
+	 *
+	 * @since 3.6.3
+	 *
+	 * @param bool $ignore_cached_hostname (optional) Don't use cached hostname when true.
+	 *
+	 * @return string Name of the hosting service or '' if no service is recognized.
+	 */
+	public static function get_host_service( $ignore_cached_hostname = false ) {
+
+		if ( ! $ignore_cached_hostname && self::$hostname ) {
+			return self::$hostname;
```

Comment:
I think using cache with long lifetime to save this hostname value will help a lot here.

Reply:
I don't think we need a longer cache than the static variable for the duration of the current request

Reply:
Ok, I thought that using cache here will be over all requests longer time and the cache is already related to the hosting itself so if customer changed the hosting company this cache won't be found.
it's only idea :D

Reply:
So, yeah, I included this as an internal short-circuit -- mainly to make unit testing easier on a static value. It's probably poorly named, since it's not really a cached value. It's probably poorly named, since it's not really a cached value.
Why use a static value or store it at all?
Currently, this is only checking for 4 hosts, but as we move more hosts into the new architecture this will become more and more expensive as we have more and more possibilities to evaluate. While not overwhelmingly expensive, there are several places in our codebase that will call it, so it seems best not to expend all the resources every time. That said, a real cached value is probably overkill.